### PR TITLE
LPS-85903: Portlet 3.0: Public render parameters that are set in the ACTION_PHASE or EVENT_PHASE must be preserved when action-url-redirect=true

### PR DESCRIPTION
**Pull request by ngriffin7a**
@shuyangzhou This PR contains 3 commits that correct some problems that were introduced in the original commit for https://issues.liferay.com/browse/LPS-85903

### CODE REVIEW

**Comment by ngriffin7a** [comment]
@shuyangzhou I reviewed the "failures unique to this pull" and I don't see how they can be related. The changes in this PR are related to Portlet 3.0 which isn't used anywhere in the system.

Please review when you get an opportunity. Thanks, Neil

**Comment by shuyangzhou** [comment]
@Preston-Crary take this one too, the URL handling logic looks heavy, please work with @ngriffin7a to find a way to simplify it. Thanks!

File: portal-kernel/src/com/liferay/portal/kernel/portlet/PortletContainerUtil.java [hunk 1]
```diff
@@ -157,7 +158,62 @@ public static void processAction(
...
+								String[] renderURLParams = renderURLQuery.split(
+									"[&]");
+
+								for (String nameValuePair : renderURLParams) {
+									String name = nameValuePair.substring(
+										0, nameValuePair.indexOf("="));
+
+									if (name.startsWith(
+											publicRenderParameterNamespace) &&
+										!query.contains(name)) {
```
Comment (Preston-Crary) [inline portal-kernel/src/com/liferay/portal/kernel/portlet/PortletContainerUtil.java]: !query.contains(name), looks wrong to me.

What if we have something like:
renderURLQuery = ?p_r_p_name=value1
query = p_r_p_name2=value2

Then "name" is contained in query even though it's not a parameter.  Also you have a null check on query but in the loop it doesn't have a null check, so either there's a possible NPE or a useless null check.

Comment (Preston-Crary) [reply portal-kernel/src/com/liferay/portal/kernel/portlet/PortletContainerUtil.java]: Also note the contains could be wrong the other way, could be something like p_r_p_1name.

Comment (ngriffin7a) [reply portal-kernel/src/com/liferay/portal/kernel/portlet/PortletContainerUtil.java]: @Preston-Crary Good catch. I'll make a correction and get back to you.

Comment (ngriffin7a) [reply portal-kernel/src/com/liferay/portal/kernel/portlet/PortletContainerUtil.java]: See https://github.com/Preston-Crary/liferay-portal/pull/614

**Pull request closed without merging**